--- old/gvsbuild/projects/gstreamer.py	2025-11-11 13:00:00.0 +0000
+++ old/gvsbuild/projects/gstreamer.py	2025-11-11 13:00:00.0 +0000
@@ -306,3 +306,31 @@
     def build(self):
         Meson.build(self)
         self.install(r".\COPYING share\doc\gst-rtsp-server")
+
+
+@project_add
+class GstPluginGtk4(Tarball, Meson):
+    def __init__(self):
+        Project.__init__(
+            self,
+            "gst-plugin-gtk4",
+            version="0.14.3",
+            repository="https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs",
+            archive_url="https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/archive/{version}/gst-plugins-rs-{version}.tar.gz?ref_type=tags",
+            archive_filename="gst-plugins-rs-{version}.tar.gz",
+            hash="0609b2459ec29f4678edd98e69b6b0a473ef4a303d60645260245dbc23f75167",
+            dependencies=[
+                "meson",
+                "cargo",
+                "gst-plugins-base",
+                "gst-plugins-bad",
+                "gtk4",
+            ],
+        )
+        self.add_param("-Dgtk4=enabled")
+        self.add_param("--auto-features=disabled")
+
+    def build(self):
+        self.builder.exec_cargo("install cargo-c --locked")
+        Meson.build(self)
+        self.install(r".\LICENSE-MPL-2.0 share\doc\gst-plugin-gtk4")
